Rakefile

```diff
@@ -1,14 +1,15 @@
 # -*- encoding: utf-8 -*-
 # frozen_string_literal: true
 
-$:.unshift File.expand_path("../lib", __FILE__)
+$:.unshift File.expand_path(File.join("..", "lib"), __FILE__)
 require "shellwords"
 require "benchmark"
 
-RUBYGEMS_REPO = if `cd .. && git remote --verbose 2>/dev/null` =~ /rubygems/i
-  File.expand_path("..")
+NULL_DEVICE = (Gem.win_platform? ? "NUL" : "/dev/null")
+RUBYGEMS_REPO = if `git -C #{File.expand_path(File.join("..", ".."), __FILE__)} remote --verbose 2> #{NULL_DEVICE}` =~ /rubygems/i
+  File.expand_path(File.join("..", ".."), __FILE__)
 else
-  File.expand_path("tmp/rubygems")
+  File.expand_path(File.join("..", "tmp", "rubygems"), __FILE__)
```

Comment:
expanding relative to `__FILE__` in a `Rakefile` isn't necessary, rake chdirs to the directory containing the rakefile